### ISSUE TITLE
[Feature request] Fine-Grained Access Controls for Ad-Hoc queries

### ISSUE THREAD

**Issue by chunyong-lin** [kolide/fleet#2077]
Fleet server currently provisions only two roles within the web application: an administrator and a non-administrator. And both roles has permission to Ad-Hoc query against any endpoints. 

In some cases, there will be different teams access the Fleet server and we would like to have more comprehensive authorization to allow certain user groups to query certain (group of) endpoints.

**Comment by benbasscom** [comment]
This feature, as well as preventing some users from creating packs, but allowing ad-hoc queries would be very useful as well.

**Comment by jalseth** [comment]
Agreed. I'd take this a step further and add the ability to create custom user groups that can have X Y and/or Z privileges inside of the Fleet UI and API. It would be even better if these user groups could be linked to attributes from SAML.
